controllers/networkplugins_controller.go

```diff
@@ -17,35 +17,480 @@ limitations under the License.
+	if multusConfig.Namespace != "" {
+		config["Namespace"] = multusConfig.Namespace
+	} else {
+		config["Namespace"] = DEFAULT_NAMESPACE
+	}
+
+	if multusConfig.MultusImage != "" {
+		config["MultusImage"] = multusConfig.MultusImage
+	} else {
+		config["MultusImage"] = MULTUS_IMAGE
+	}
+
+	t, err := template.ParseFiles(TEMPLATE_DIR + "multus/multus.yaml")
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := createFile(config, t, outputDir+"multus.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (multusConfig *MultusT) ApplyTemplate(outputDir string) error {
+	fmt.Println("Applying Multus")
+	return nil
+}
+
+func (whereaboutsConfig *WhereaboutsT) ParseTemplate(outputDir string) error {
+	config := make(map[string]interface{})
+	fmt.Println("Rendering Whereabouts...")
+	if whereaboutsConfig.Namespace != "" {
+		config["Namespace"] = whereaboutsConfig.Namespace
+	} else {
+		config["Namespace"] = DEFAULT_NAMESPACE
+	}
+
+	if whereaboutsConfig.WhereaboutsImage != "" {
+		config["WhereaboutsImage"] = whereaboutsConfig.WhereaboutsImage
+	} else {
+		config["WhereaboutsImage"] = WHEREABOUTS_IMAGE
+	}
+
+	t, err := template.ParseFiles(TEMPLATE_DIR + "whereabouts/whereabouts.yaml")
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := createFile(config, t, outputDir+"whereabouts.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (whereaboutsConfig *WhereaboutsT) ApplyTemplate(outputDir string) error {
+	fmt.Println("Applying Whereabouts")
+	return nil
+}
+
+func (sriovConfig *SriovT) ParseTemplate(outputDir string) error {
+	config := make(map[string]interface{})
+	fmt.Println("Rendering Sriov...")
+
+	if sriovConfig.Namespace != "" {
+		config["Namespace"] = sriovConfig.Namespace
+	} else {
+		config["Namespace"] = DEFAULT_NAMESPACE
+	}
+
+	if sriovConfig.SriovCniImage != "" {
+		config["SriovCniImage"] = sriovConfig.SriovCniImage
+	} else {
+		config["SriovCniImage"] = SRIOV_CNI_IMAGE
+	}
+
+	if sriovConfig.SriovDpImage != "" {
+		config["SriovDpImage"] = sriovConfig.SriovDpImage
+	} else {
+		config["SriovDpImage"] = SRIOV_DP_IMAGE
+	}
+
+	// Apply the SRIOV CNI
+	t, err := template.ParseFiles(TEMPLATE_DIR + "sriov/sriov-cni.yaml")
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := createFile(config, t, outputDir+"sriov-cni.yaml"); err != nil {
+		return err
+	}
+
+	// Apply the SRIOV Device Plugin
+	t, err = template.ParseFiles(TEMPLATE_DIR + "sriov/sriov-deviceplugin.yaml")
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := createFile(config, t, outputDir+"sriov-deviceplugin.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sriovConfig *SriovT) ApplyTemplate(outputDir string) error {
+	fmt.Println("Applying SRIOV")
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) parsePlugin(plugin ApplyPlugin, mode string) error {
```

Comment:
Also here the function does not really reflect what it is doing. It is unclear how `mode` relates to `parse`. If there aren't many expected modes I'd just split this function into two functions, one for 'delete' and one for 'create'